### PR TITLE
adds TaylorGP for competition

### PR DESCRIPTION
**Pull request by lacava**
cc @KGAE-CUP

Competition Checklist:
- [x] title of this PR is meaningful, i.e. "method X for comp"
- [x] A folder has been added to `submission/` with a meaningful name corresponding to your method name.
- The added folder includes these elements:
    - [x] `metadata.yml` (**required**): A file describing your submission, following the descriptions in `example/metadata.yml`.  
    - [x] `regressor.py` (**required**): a Python file that defines your method, named appropriately. See [submission/feat-example/regressor.py](https://github.com/cavalab/srbench/blob/Competition2022/submission/feat-example/regressor.py) for complete documentation.  It contains:
        - [x]  `est`: a sklearn-compatible `Regressor` object. 
        - [x]  `model(X_Y=None)`: a function that returns a [**sympy-compatible**](https://www.sympy.org) string specifying the final model. It can optionally take the training data as an input argument. See [guidance below](###-returning-a-sympy-compatible-model-string). 
        - []  `eval_kwargs` (optional): a dictionary that can specify method-specific arguments to `evaluate_model.py`.
        
    - [] `LICENSE` *(optional)* A license file
    - [x] `environment.yml` *(optional)*: a [conda environment file](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-from-an-environment-yml-file) that specifies dependencies for your submission. 
    - [] `install.sh` *(optional)*: a bash script that installs your method. 
    - [x] additional files *(optional)*: you may include a folder containing the code for your method in the submission. 

I have verified that:

- [x] install scripts do not require sudo permissions. 
- [x] if pulled remotely, the source code is a fixed version (i.e., rerunning `install.sh` shouldn't pulll a different version of the code when run multiple times.) 

Refer to the [competition guide]( https://cavalab.org/srbench/competition-guide/ ) if you are unsure about any steps. 
If you don't find an answer, ping us!

### CODE REVIEW

**Comment by lacava** [comment]
@KGAE-CUP please provide an `install.sh` with something simple like 

```bash
cd taylorGP
python setup.py install
```

Otherwise `python setup.py install` won't be triggered anywhere. Hopefully will work after that.

**Comment by KGAE-CUP** [comment]
Thank you for your valuable comments! I saw that there are many ways to submit code in competition requirements. To replace the "install. sh" method, I chose environmen.yml  to install the virtual environment locally through "conda env create - f environment. yml". Do you have any suggestions on this?
Thank you for your letter.


> @KGAE-CUP please provide an `install.sh` with something simple like
> 
> ```shell
> cd taylorGP
> python setup.py install
> ```
> 
> Otherwise `python setup.py install` won't be triggered anywhere. Hopefully will work after that.

Thank you for your valuable comments! I saw that there are many ways to submit code in competition requirements. To replace the "install. sh" method, I chose environmen.yml  to install the virtual environment locally through "conda env create - f environment. yml". Do you have any suggestions on this?
Thank you for your letter.

**Comment by lacava** [comment]
> @lacava Thank you for your valuable comments! I saw that there are many ways to submit code in competition requirements. To replace the "install. sh" method, I chose environmen.yml  to install the virtual environment locally through "conda env create - f environment. yml". Do you have any suggestions on this?
> Thank you for your letter.
> 

Unless taylorGP is a conda package, you will need to both 1) create the env via environment.yml and 2) call 'python setup.py install' in the package directory. the install.sh file would accomplish 2).

this is what i understand from the .travis.yml file in your submission - i left a comment on the relevant line. you should confirm this is how to install the package.

File: submission/TaylorGP/metadata.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+name: TaylorGP # name of the submission /method
+description: | # anything you'd like here to describe the method.
+  Taylor Genetic Programming for Symbolic Regression.
+  The Taylor features mined by Taylor polynomial that fits given dataset are used to guide genetic programming.
+  
+paperName: "Taylor Genetic Programming for Symbolic Regression" 
+  GECCO 2022. preprint: https://kgae-cup.github.io/TaylorGP/
+
+  Really hope I win! jk
+url: http://cavalab.org/feat # a link to the project
```
Comment (lacava) [inline submission/TaylorGP/metadata.yml:13]: change the URL

**Review by lacava — COMMENTED** [review]
looks good @KGAE-CUP ! Once you fix the metadata.yaml, should be good to merge.

File: submission/TaylorGP/metadata.yml (hunk 1 above)
Comment (KGAE-CUP) [reply submission/TaylorGP/metadata.yml]: Sorry, I forgot to modify the url in metadata.yml. Thank you for your reminder. Now I have modified it.

Comment (KGAE-CUP) [reply submission/TaylorGP/metadata.yml]: > change the URL

 Sorry, I forgot to modify the url in metadata.yml. Thank you for your reminder. Now I have modified it.

**Comment by KGAE-CUP** [comment]
@lacava Hello lcava, I found that the "install. sh" in the "experiment/methods/TaylorGP" folder is empty, which is not the same as the install. sh in the "submission/TaylorGP" folder, will this cause my algorithm fail to test datasets?

**Comment by lacava** [comment]
hi @KGAE-CUP no, you should be all set. I will let you know if I run into any issues.

**Pull request merged**